.github/workflows/cd.yml

```diff
@@ -3,10 +3,12 @@ name: NPM publish CD workflow
 on:
   push:
     branches: [ master ]
+  workflow_dispatch:
 
 jobs:
   build:
-    if: "!startsWith(github.event.head_commit.message, '[ci skip]')"
+    # if triggered by push, ignore commits that workflow pushed itself
+    if: "github.event_name === 'workflow_dispatch' || !startsWith(github.event.head_commit.message, '[ci skip]')"
```

Comment:
Not sure exactly how `github.event_name` works, but it looks like maybe this logic is wrong? Your comment says "ignore commits that workflow pushed itself" but the check is `github.event_name === 'workflow_dispatch'`.

Reply:
yeah the condition is reversed which makes it rather unreadable.. I didn't find an easy syntax to introduce intermediate variable though... is it better with additional comment?

Reply:
Sure, so long as it's correct.